src/ast/nodes/IfStatement.ts

```diff
@@ -45,33 +50,41 @@ export default class IfStatement extends StatementBase {
 	}
 
 	render(code: MagicString, options: RenderOptions) {
-		const testValue = this.test.getValue();
+		const testValue = this.hasUnknownTestValue ? UNKNOWN_VALUE : this.test.getValue();
 		if (
 			!this.context.treeshake ||
 			this.test.included ||
 			(testValue ? this.alternate !== null && this.alternate.included : this.consequent.included)
 		) {
-			super.render(code, options);
+			this.test.render(code, options);
+			if (this.consequent.included) {
+				this.consequent.render(code, options);
+			} else {
+				code.overwrite(this.consequent.start, this.consequent.end, '{}');
```

Comment:
Can we not keep this as `;` as the tests are currently? It still looks much nicer as an empty statement to me.

Reply:
I guess you are right.